packages/dht/src/dht/DhtPeer.ts

```diff
@@ -0,0 +1,106 @@
+import { IDhtRpcClient } from '../proto/DhtRpc.client'
+import { ClosestPeersRequest, PeerDescriptor, PingRequest, RouteMessageWrapper } from '../proto/DhtRpc'
+import { v4 } from 'uuid'
+import { PeerID } from '../helpers/PeerID'
+import { jsFormatPeerDescriptor } from '../helpers/common'
+import { DhtRpcOptions } from '../rpc-protocol/DhtRpcOptions'
+import { RouteMessageParams } from './DhtNode'
+import { Logger } from '../helpers/Logger'
+
+const logger = new Logger(module)
+
+export class DhtPeer {
+    private static counter = 0
+    
+    public readonly peerId: PeerID
+
+    public get id(): Uint8Array {
+        return this.peerId.value
+    }
+
+    private lastContacted: number
+    private peerDescriptor: PeerDescriptor
+    public vectorClock: number
```

Comment:
What is the purpose of this field? It is incremented in this class and in `Contact`, but not analyzed anywhere?

Reply:
Many of the seemingly unused fields here are required and used by the k-bucket library

Reply:
If it is needed by external component, we should have an interface which this class implements?